diff -up pix-2.8.9/pix/gth-browser.c.omv~ pix-2.8.9/pix/gth-browser.c
diff -up pix-2.8.9/pix/gth-folder-tree.c.omv~ pix-2.8.9/pix/gth-folder-tree.c
--- pix-2.8.9/pix/gth-folder-tree.c.omv~	2023-05-18 05:48:18.084283717 +0200
+++ pix-2.8.9/pix/gth-folder-tree.c	2023-05-18 05:50:12.556408015 +0200
@@ -315,7 +315,7 @@ icon_renderer_func (GtkTreeViewColumn *c
                     GtkCellRenderer   *cell,
                     GtkTreeModel      *model,
                     GtkTreeIter       *iter,
-                    gpointer          *data)
+                    gpointer          data)
 {
     GdkPixbuf *pixbuf;
     GthFolderTree *folder_tree = GTH_FOLDER_TREE (data);
diff -up pix-2.8.9/pix/gth-location-chooser.c.omv~ pix-2.8.9/pix/gth-location-chooser.c
--- pix-2.8.9/pix/gth-location-chooser.c.omv~	2023-05-18 05:50:30.340582510 +0200
+++ pix-2.8.9/pix/gth-location-chooser.c	2023-05-18 05:50:35.772635799 +0200
@@ -577,7 +577,7 @@ icon_renderer_func (GtkCellLayout *cell_
                     GtkCellRenderer   *cell,
                     GtkTreeModel      *model,
                     GtkTreeIter       *iter,
-                    gpointer          *data)
+                    gpointer          data)
 {
     GdkPixbuf *pixbuf;
 
diff -up pix-2.8.9/pix/gth-window.c.omv~ pix-2.8.9/pix/gth-window.c
--- pix-2.8.9/pix/gth-window.c.omv~	2023-05-18 05:46:41.952337800 +0200
+++ pix-2.8.9/pix/gth-window.c	2023-05-18 05:46:48.744404692 +0200
@@ -197,7 +197,7 @@ gth_window_real_set_current_page (GthWin
 }
 
 /* extern */
-gint *global_ui_scale;
+gint global_ui_scale;
 
 static void
 ui_scale_changed (GtkWidget *widget)
diff -up pix-2.8.9/pix/gtk-utils.h.omv~ pix-2.8.9/pix/gtk-utils.h
--- pix-2.8.9/pix/gtk-utils.h.omv~	2023-05-18 05:46:54.675463097 +0200
+++ pix-2.8.9/pix/gtk-utils.h	2023-05-18 05:46:57.985495688 +0200
@@ -44,7 +44,7 @@ typedef struct {
 	GCallback       callback;
 } GthActionEntryExt;
 
-extern gint *global_ui_scale;
+extern gint global_ui_scale;
 
 void            _gtk_action_group_add_actions_with_flags   (GtkActionGroup   *action_group,
 							    const GthActionEntryExt
